plugins/mongodb/mongodbembed.cpp

```diff
@@ -0,0 +1,2366 @@
+    }
+
+    /**
+     * @brief Builds an ECL row from the result documents of MongoDB query
+     * 
+     * @param _resultAllocator Used for building the ECL row by the engine.
+     * @return byte* Pointer to ECL row handed back to the engine.
+     */
+    byte * MongoDBEmbedFunctionContext::getRowResult(IEngineRowAllocator * _resultAllocator)
+    {
+        Owned<MongoDBRowStream> mdbRowStream;
+        mdbRowStream.setown(new MongoDBRowStream(_resultAllocator, query));
+        return (byte *)mdbRowStream->nextRow();
+    }
+
+    // MongoDB does not return scalar values and both the order of documents and fields in those documents are not guarenteed.
+    // Therefore the ECL user can have no expectation of what the output will be.
+    size32_t MongoDBEmbedFunctionContext::getTransformResult(ARowBuilder & rowBuilder)
+    {
+        UNIMPLEMENTED_X("MongoDB Transform Result");
+        return 0;
+    }
+
+    bool MongoDBEmbedFunctionContext::getBooleanResult() 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type BOOLEAN");
+        return false;
+    }
+    
+    __int64 MongoDBEmbedFunctionContext::getSignedResult() 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type SIGNED INTEGER");
+        return 0;
+    }
+    
+    unsigned __int64 MongoDBEmbedFunctionContext::getUnsignedResult() 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type UNSIGNED INTEGER");
+        return 0;
+    }
+
+    void MongoDBEmbedFunctionContext::getDataResult(size32_t &len, void * &result) 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type DATA");
+    }
+    
+    double MongoDBEmbedFunctionContext::getRealResult() 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type REAL");
+        return 0.0;
+    }
+    
+    void MongoDBEmbedFunctionContext::getStringResult(size32_t &chars, char * &result) 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type STRING");
+    }
+    
+    void MongoDBEmbedFunctionContext::getUTF8Result(size32_t &chars, char * &result) 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type UTF8");
+    }
+    
+    void MongoDBEmbedFunctionContext::getUnicodeResult(size32_t &chars, UChar * &result) 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type UNICODE");
+    }
+    
+    void MongoDBEmbedFunctionContext::getDecimalResult(Decimal &value) 
+    {
+        UNIMPLEMENTED_X("MongoDB Scalar Return type DECIMAL");
+    }
+
+    /**
+     * @brief Compiles the embedded script and stores it in the MongoDBQuery object.
+     * 
+     * @param chars Length of the embedded script.
+     * @param script Pointer to the script.
+     */
+    void MongoDBEmbedFunctionContext::compileEmbeddedScript(size32_t chars, const char *script)
+    {
+        if (script && *script) 
+        {
+            // Incoming script is not necessarily null terminated. Note that the chars refers to utf8 characters and not bytes.
+            size32_t len = rtlUtf8Size(chars, script);
```

Comment:
minor: size is a better name than len (len is normally used for the number of characters/codepoints)

Reply:
Changed